nifi-commons/nifi-data-provenance-utils/src/main/java/org/apache/nifi/provenance/AsyncQuerySubmission.java

```diff
@@ -82,4 +90,8 @@ public Query getQuery() {
     public StandardQueryResult getResult() {
         return queryResult;
     }
+
+    public void addQueryExecution(Future<?> execution) {
```

Comment:
Is this method only on AsyncQuerySubmission (versus the parent interface) because adding an execution only applies to async queries? I figure it is but wanted to make sure. If other implementations would benefit from this method, it might be better as an default interface method on the parent that throws a MethodNotSupported or something. But if it really only applies in this setting, I'm fine with that too.